src/wp-includes/blocks.php

```diff
@@ -757,6 +757,44 @@ function get_hooked_blocks() {
 	return $hooked_blocks;
 }
 
+function insert_hooked_blocks( $relative_position, &$anchor_block, $hooked_blocks, $context ) {
+	$anchor_block_type  = $anchor_block['blockName'];
+	$hooked_block_types = isset( $hooked_blocks[ $anchor_block_type ][ $relative_position ] )
+		? $hooked_blocks[ $anchor_block_type ][ $relative_position ]
+		: array();
+
+	$markup = '';
+	/**
+	 * Filters the list of hooked block types for a given anchor block type and relative position.
+	 *
+	 * @since 6.4.0
+	 *
+	 * @param string[]                $hooked_block_types  The list of hooked block types.
+	 * @param string                  $relative_position   The relative position of the hooked blocks.
+	 *                                                     Can be one of 'before', 'after', 'first_child', or 'last_child'.
+	 * @param string                  $anchor_block_type   The anchor block type.
+	 * @param WP_Block_Template|array $context             The block template, template part, or pattern that the anchor block belongs to.
+	 */
+	$hooked_block_types = apply_filters( 'hooked_block_types', $hooked_block_types, $relative_position, $anchor_block_type, $context );
+	foreach ( $hooked_block_types as $hooked_block_type ) {
+		if (
+			! isset( $anchor_block['attrs']['metadata']['ignoredHookedBlocks'] ) ||
+			! in_array( $hooked_block_type, $anchor_block['attrs']['metadata']['ignoredHookedBlocks'] )
+		) {
+			$markup .= get_comment_delimited_block_content( $hooked_block_type, array(), '' );
+		}
+
+		// TODO: The following is only needed for the REST API endpoint.
+		// Ideally, the code should thus be moved into the controller.
+		if ( ! isset( $anchor_block['attrs']['metadata']['ignoredHookedBlocks'] ) ) {
```

Comment:
Is there any issue with running it always? It's a simple operation, so I don't expect it would impact the front end. I would be in favor of keeping it simple unless I miss something.

Reply:
No issue, other than it adds a few bytes per block (and it _might_ be somewhat confusing for people). I just thought if we revisit https://github.com/WordPress/wordpress-develop/pull/5514 (which requires different treatment of frontend and REST API), we might also look into this. Definitely okay to leave it in though.